src/engine/source/api/src/catalog/catalog.cpp

```diff
@@ -323,33 +329,64 @@ std::optional<base::Error> Catalog::putResource(const Resource& item, const std:
+    {
+        const auto formatterIt = outFormat.find(format);
+        if (formatterIt == outFormat.end())
+        {
+            return base::Error {fmt::format("Formatter was not found for format '{}'", Resource::formatToStr(format))};
+        }
+
+        const auto formatResult = formatterIt->second(content);
+        if (std::holds_alternative<base::Error>(formatResult))
+        {
+            return base::Error {fmt::format("JSON object could not be created from '{} {}': {}",
+                                           Resource::formatToStr(format),
+                                           name.fullName(),
+                                           std::get<base::Error>(formatResult).message)};
+        }
+
+        return std::get<std::string>(formatResult);
+    };
+
+    const auto storeResult = store::utils::get(m_store, resource.m_name, true);
     if (std::holds_alternative<base::Error>(storeResult))
     {
-        return base::Error {fmt::format("Content '{}' could not be obtained from store: {}",
-                                        resource.m_name.fullName(),
-                                        std::get<base::Error>(storeResult).message)};
+        return base::Error {std::get<base::Error>(storeResult).message};
     }
 
-    const auto contentJson = std::get<json::Json>(storeResult);
-    // Format the content to the expected output format
-    const auto formatterIt = m_outFormat.find(resource.m_format);
-    if (formatterIt == m_outFormat.end())
+    const auto& content = std::get<json::Json>(storeResult);
+
+    if (resource.m_format == Format::json || resource.m_type == Type::collection || resource.m_type == Type::policy)
     {
-        return base::Error {
-            fmt::format("Formatter was not found for format '{}'", Resource::formatToStr(resource.m_format))};
+        if (content.exists("/original"))
+        {
+            auto original = content.getJson("/json");
+            if (original)
+            {
+                return formatContent(std::move(original.value()));
+            }
+            else
+            {
+                return base::Error {"Could not get the original content from the store"};
+            }
+        }
+
+        return formatContent(content);
     }
-    const auto formatResult = formatterIt->second(contentJson);
-    if (std::holds_alternative<base::Error>(formatResult))
+
+    if (Resource::formatToStr(Format::json) == content.getString("/format").value())
     {
-        return base::Error {fmt::format("JSON object could not be created from '{} {}': {}",
-                                        Resource::formatToStr(resource.m_format),
-                                        resource.m_name.fullName(),
-                                        std::get<base::Error>(formatResult).message)};
+        if (Resource::formatToStr(Format::yaml) == Resource::formatToStr(resource.m_format))
```

Comment:
unify the `if` confitions:
```suggestion
        if (Resource::formatToStr(Format::json) == Resource::formatToStr(content.getString("/format").value())
        && Resource::formatToStr(Format::yaml) == Resource::formatToStr(resource.m_format))
```